### PR TITLE
Add asserts for ref frames update with frame reordering

### PR DESCRIPTION
**Pull request by ycho**

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -257,8 +257,10 @@ impl Context {
         self.fi.ref_frames[i] = if i == second_ref_frame - LAST_FRAME {
           (slot_idx as u64 + if lvl == 0 { 6 * group_src_len } else { group_src_len >> lvl }) & 7
         } else if i == ref_in_previous_group - LAST_FRAME {
+          assert!((slot_idx as u64 - group_src_len) >= 0);
           (slot_idx as u64 - group_src_len) & 7
         } else {
+          assert!((slot_idx as u64 - (group_src_len >> lvl)) >= 0);
```
Comment (fbossen) [inline src/api.rs]: Am seeing the following during build: "warning: comparison is useless due to type limits"
Note that the computation below uses &7 to deal with cases where wraparound occurs.

File: src/api.rs [hunk 2]
```diff
@@ -257,8 +257,10 @@ impl Context {
         self.fi.ref_frames[i] = if i == second_ref_frame - LAST_FRAME {
           (slot_idx as u64 + if lvl == 0 { 6 * group_src_len } else { group_src_len >> lvl }) & 7
         } else if i == ref_in_previous_group - LAST_FRAME {
+          assert!((slot_idx as u64 - group_src_len) >= 0);
```
Comment (fbossen) [inline src/api.rs]: Am seeing the following during build: "warning: comparison is useless due to type limits"
Note that the computation below uses &7 to deal with cases where wraparound occurs.

Comment (lu-zero) [reply src/api.rs]: In that case you should explicitly use [wrapping_sub()](https://doc.rust-lang.org/std/primitive.u64.html#method.wrapping_sub).

**Comment by ycho** [comment]
Since the issue has been resolved the other way by fbossen, #647 , closing this.

**Pull request closed without merging**